backend/src/controller/chatController.ts

```diff
@@ -162,11 +163,7 @@ async function handleChatWithDefenceDetection(
 
 	// if blocked, restore original chat history and add user message to chat history without completion
 	const updatedChatHistory = combinedDefenceReport.isBlocked
-		? pushMessageToHistory(chatHistory, {
-				completion: null,
-				chatMessageType: CHAT_MESSAGE_TYPE.USER,
-				infoMessage: message,
-		  })
+		? chatHistoryWithNewUserMessages
```

Comment:
I've lost track of this PR a touch! Looks like you're keeping the transformed message now even if it was blocked, so I'm assuming that's a deliberate change.

Reply:
Ah, it was not so deliberate. I want the new messages in the history, but I do not want the completion there. I shall change that now

Reply:
So now we need a version of `type ChatUserTransformedMessage` which doesn't include a completion. I was in two minds about whether I should just make the completion optional, or to make a whole new type that doesn't have the completion. I went with the former for lightweightness, but very happy to try the latter.